docs/tutorials/getting-started-with-csharp/working-with-linq.md

```diff
@@ -0,0 +1,510 @@
+            (this IEnumerable<T> first, IEnumerable<T> second)
+        {
+            // implementation coming.
+        }
+    }
+}
+```
+
+You will be enumerating both sequences at once, interleaving the elements,
+and creating one object.  Writing a LINQ method that works with two
+sequences requires that you understand how `IEnumerable` works.
+
+The `IEnumerable` interface has one method: `GetEnumerator()`. The object
+returned by `GetEnumerator()` has a method to move to the next element,
+and a property that retrieves the current element in the sequence. You
+will use those two members to enumerate the collection and return the
+elements. This Interleave method will be an iterator method, so instead
+of building a collection and returning the collection, you'll use the 
+`yield return` syntax shown above. 
+
+Here's the implementation of that method:
+
+```cs
+public static IEnumerable<T> InterleaveSequenceWith<T>
+    (this IEnumerable<T> first, IEnumerable<T> second)
+{
+    var firstIter = first.GetEnumerator();
+    var secondIter = second.GetEnumerator();
+    while (firstIter.MoveNext() && secondIter.MoveNext())
+    {
+        yield return firstIter.Current;
+        yield return secondIter.Current;
+    }
+}
+```
+
+Now that you've written this method, go back to the `Main` method
+and shuffle the deck once:
+
+```cs
+public static void Main(string[] args)
+{
+    var startingDeck = from s in Suits()
+                       from r in Ranks()
+                       select new { Suit = s, Rank = r };
+    foreach (var c in startingDeck)
+        Console.WriteLine(c);
+        
+    var top = startingDeck.Take(26);
+    var bottom = startingDeck.Skip(26);
+    
+    var shuffle = top.InterleaveSequenceWith(bottom);
+    foreach (var c in shuffle)
+        Console.WriteLine(c);
+}
+```
+
+# Comparisons
+
+Let's see how many shuffles it taks to set the deck back to its
+original order. You'll need to write a method that determines if
+two sequences are equal. After you have that method, you'll need to
+plac the code that shuffles the deck in a loop, and check tosee when
+the deck is back in order.
+
+Writing a method to determine if the two sequences are equal should
+be straightforward. It's a similar structure to the method you wrote
+to shuffle the deck. Only this time, instead of yield returning each
+element, you'll compare the matching elements of each sequence. When
+the entire sequence has been enumerated, if every element matches,
+the sequences are the same:
+
+```cs
+public static bool SequenceEquals<T>(this IEnumerable<T> first, IEnumerable<T> second)
+{
+    var firstIter = first.GetEnumerator();
+    var secondIter = second.GetEnumerator();
+    while (firstIter.MoveNext() && secondIter.MoveNext())
+    {
+        if (!firstIter.Current.Equals(secondIter.Current))
+            return false;
+    }
+    return true;
+}
+```
+
+This shows a second Linq idiom: terminal methods. They take a sequence as input
```

Comment:
For me this was not easy to follow. It might help if you specify the terminal method and show it in operation in a way that shows how it acts to terminate a query - as it is it looks like nothing special since it appears to be called like an ordinary method , so I have to imagine maybe a query expression where it plays the special terminating role.